File: src/main/java/com/github/nsc/de/shake/parser/node/expression/ModNode.java
package com.github.nsc.de.shake.parser.node.expression;

import com.github.nsc.de.shake.parser.node.ValuedNode;

public class ModNode extends ExpressionNode {
    public ModNode(ValuedNode left, ValuedNode right, int operatorIndex) {
        super(left, right, operatorIndex);
    }

    @Override
    public char getOperator() {
        return '%';
    }
}

File: shake/compiler/parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/expression/ModNode.kt
package io.github.shakelang.shake.parser.node.expression

import io.github.shakelang.shake.parser.node.ValuedNode
import io.github.shakelang.parseutils.characters.position.PositionMap

class ModNode(map: PositionMap, left: ValuedNode, right: ValuedNode, operatorPosition: Int) :
    ExpressionNode(map, left, right, operatorPosition) {
    override val operator: Char
        get() = '%'

    override fun toJson(): Map<String, *> = mapOf("name" to "ModNode", "left" to left, "right" to right)
}